docs/how-to/hip_runtime_api/multi_device.rst

```diff
@@ -0,0 +1,423 @@
+appropriate device, you ensure that the computational tasks are directed to the
+correct GPU, optimizing performance and resource utilization.
+
+.. code-block:: cpp
+
+    #include <hip/hip_runtime.h>
+    #include <iostream>
+
+    #define HIP_CHECK(expression)                \
+    {                                            \
+        const hipError_t status = expression;    \
+        if (status != hipSuccess) {              \
+            std::cerr << "HIP error " << status  \
+                    << ": " << hipGetErrorString(status) \
+                    << " at " << __FILE__ << ":" \
+                    << __LINE__ << std::endl;  \
+            exit(status);                        \
+        }                                        \
+    }
+
+    __global__ void simpleKernel(double *data)
+    {
+        int idx = blockIdx.x * blockDim.x + threadIdx.x;
+        data[idx] = idx * 2.0;
+    }
+
+    int main()
+    {
+        double* deviceData0;
+        double* deviceData1;
+        size_t  size = 1024 * sizeof(*deviceData0);
+
+        int deviceId0 = 0;
+        int deviceId1 = 1;
+
+        // Set device 0 and perform operations
+        HIP_CHECK(hipSetDevice(deviceId0)); // Set device 0 as current
+        HIP_CHECK(hipMalloc(&deviceData0, size)); // Allocate memory on device 0
+        simpleKernel<<<1000, 128>>>(deviceData0); // Launch kernel on device 0
+        HIP_CHECK(hipDeviceSynchronize());
+
+        // Set device 1 and perform operations
+        HIP_CHECK(hipSetDevice(deviceId1)); // Set device 1 as current
+        HIP_CHECK(hipMalloc(&deviceData1, size)); // Allocate memory on device 1
+        simpleKernel<<<1000, 128>>>(deviceData1); // Launch kernel on device 1
+        HIP_CHECK(hipDeviceSynchronize());
+
+        // Copy result from device 0
+        double hostData0[1024];
+        HIP_CHECK(hipSetDevice(deviceId0));
+        HIP_CHECK(hipMemcpy(hostData0, deviceData0, size, hipMemcpyDeviceToHost));
+
+        // Copy result from device 1
+        double hostData1[1024];
+        HIP_CHECK(hipSetDevice(deviceId1));
+        HIP_CHECK(hipMemcpy(hostData1, deviceData1, size, hipMemcpyDeviceToHost));
+
+        // Display results from both devices
+        std::cout << "Device 0 data: " << hostData0[0] << std::endl;
+        std::cout << "Device 1 data: " << hostData1[0] << std::endl;
+
+        // Free device memory
+        HIP_CHECK(hipFree(deviceData0));
+        HIP_CHECK(hipFree(deviceData1));
+
+        return 0;
+    }
+
+
+Stream and event behavior
+===============================================================================
+
+In a multi-device system, streams and events are essential for efficient
+parallel computation and synchronization. Streams enable asynchronous task
+execution, allowing multiple devices to process data concurrently without
+blocking one another. Events provide a mechanism for synchronizing operations
```

Comment:
I think we should indicate that events allow synchronization across streams and across devices.

Reply:
Thanks! Updated.